src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,89 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private static final int ARRAY_SIZE = 16;
+    private static final int MULTIPLIER = 2;
+    private Node<K, V>[] table;
+    private int size;
+
+    public MyHashMap() {
+        table = (Node<K, V>[]) new Node[ARRAY_SIZE];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        int hash = hash(key);
+        int index = hash & (table.length - 1);
+        Node<K, V> node = table[index];
+        while (node != null) {
+            if (Objects.equals(key, node.key)) {
+                node.value = value;
+                return;
+            }
+            node = node.next;
+        }
+        Node<K, V> newNode = new Node<K, V>(key, value, hash, null);
+        newNode.next = table[index];
```

Comment:
why you are doing it? if you trying to put new node, you have to change "next" link for current last node and then make new one the last